src/main/java/com/github/manolo8/darkbot/core/objects/facades/QuestProxy.java

```diff
@@ -0,0 +1,323 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.swf.FlashList;
+import com.github.manolo8.darkbot.core.objects.swf.ObjArray;
+
+import eu.darkbot.api.managers.QuestAPI;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jetbrains.annotations.Nullable;
+
+public class QuestProxy extends Updatable implements QuestAPI {
+    private @Nullable QuestProxy.Quest currentQuest;
+
+    private final List<QuestProxy.QuestListItem> questItems = new ArrayList<>();
+
+    private final ObjArray questItemsArr = ObjArray.ofArrObj(true);
+
+    private @Nullable QuestProxy.Quest questGiverSelected;
+    private @Nullable QuestProxy.QuestListItem questInfoGiverSelected;
+
+    private boolean visibleQuestGiver;
+    private int tabSelected;
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            return;
+        }
+
+        long questClass = API.readMemoryPtr(address + 0x98);
+
+        long currentQuestAddr = API.readMemoryPtr(questClass + 0x28);
+        if (this.currentQuest == null) {
+            this.currentQuest = new Quest();
+        }
+        this.currentQuest.update(currentQuestAddr);
+
+        this.visibleQuestGiver = API.readMemoryBoolean(address, 0x40);
+
+        if (!visibleQuestGiver) {
+            this.questInfoGiverSelected = null;
+            this.questGiverSelected = null;
+            return;
+        }
+
+        this.tabSelected = API.readMemoryInt(address, 0x50);
+
+        long questInfoGiverSelectedAddr = API.readMemoryPtr(address + 0xA8);
+        if (this.questInfoGiverSelected == null) {
+            this.questInfoGiverSelected = new QuestListItem();
+        }
+        this.questInfoGiverSelected.update(questInfoGiverSelectedAddr);
+
+        long questGiverSelectedAddr = API.readMemoryPtr(address + 0xB0);
+        if (this.questGiverSelected == null) {
+            this.questGiverSelected = new Quest();
+        }
+        this.questGiverSelected.update(questGiverSelectedAddr);
+    }
+
+    @Override
+    public @Nullable QuestAPI.Quest getDisplayedQuest() {
+        return currentQuest == null || currentQuest.address == 0 ? null : currentQuest;
+    }
+
+    @Override
+    public @Nullable QuestAPI.Quest getSelectedQuest() {
+        return questGiverSelected == null || questGiverSelected.address == 0 ? null : questGiverSelected;
+    }
+
+    @Override
+    public @Nullable QuestAPI.QuestListItem getSelectedQuestInfo() {
+        return questInfoGiverSelected == null || questInfoGiverSelected.address == 0 ? null : questInfoGiverSelected;
+    }
+
+    @Override
+    public @Nullable List<? extends QuestAPI.QuestListItem> getCurrestQuests() {
+        questItemsArr.update(API.readMemoryPtr(address + 0x58));
+        questItemsArr.sync(questItems, QuestListItem::new);
+
+        return questItems;
+    }
```

Comment:
The implementation of this method could cause a naive plugin to cause tons of re-reading in memory, eg:
```java
for (int i = 0; i < getCurrentQuests().size(); i++) {
  if (getCurrentQuests().get(i).something()) { ... }
}
```
I know it's annoying, but this needs to happen only at most once per tick, you can wrap it arround an:
```java 
if (!questsUpdated) {
  questsUpdated = true;
  questItemsArr.update(API.readMemoryPtr(address + 0x58));
  questItemsArr.sync(questItems, QuestListItem::new);
}
return questItemsArr
```
and just set questsUpdated back to false on the tick method